lib/Runtime/Types/DictionaryTypeHandler.cpp

```diff
@@ -2193,11 +2193,19 @@ namespace Js
             SetPropertyValueInfo(info, instance, index, attributes);
         }
 
-        // Always invalidate prototype caches when we add a property.  Previously, we only did this if the current
-        // type is used as a prototype, or if the new property is also found on the prototype chain (because
-        // adding a new field doesn't create a new dictionary type).  However, if the new property is already in
-        // the cache as a missing property, we have to invalidate the prototype caches.
-        scriptContext->InvalidateProtoCaches(propertyRecord->GetPropertyId());
+        if (!IsInternalPropertyId(propertyRecord->GetPropertyId()) && ((this->GetFlags() & IsPrototypeFlag)
+            || JavascriptOperators::HasProxyOrPrototypeInlineCacheProperty(instance, propertyRecord->GetPropertyId())))
+        {
```

Comment:
Why is this condition different than the one in SimpleDictionaryTypeHandler?

Reply:
Good catch -- probably a transcription error on my part.  Fixed.

Reply:
Then of course my next question is, can both of those call a single helper function so we don't have to worry about whether they match? :)

---
In reply to: [173578136](https://github.com/Microsoft/ChakraCore/pull/4780#discussion_r173578136) [](ancestors = 173578136)

Reply:
Fixed